extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluegiga/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java

```diff
@@ -417,6 +417,11 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
             if (characteristic == null) {
                 logger.debug("BlueGiga didn't find characteristic for event {}", event);
             } else {
+            	
+            	 procedureCharacteristic.setValue(valueEvent.getValue());
+            	// Notify the user of the updated value
+            	notifyListeners(BluetoothEventType.CHARACTERISTIC_UPDATED, procedureCharacteristic);
```

Comment:
Why did you move this code up here? This changes the order of the notifications and imho it was just fine before.

Reply:
I thought the **CHARACTERISTIC_UPDATED** event should send the `procedureCharacteristic` before it is set to null [here](https://github.com/kaikreuzer/smarthome/pull/8/files#diff-a7b7ef841bb09679f68c5cab00300b0bR426) as I used the characteristic in the `onCharacteristicUpdate` method in my binding initially. I don't override the `onCharacteristicUpdate` in my binding anymore and I reverted the change, but I'm still curious why would the code [here](https://github.com/kaikreuzer/smarthome/pull/8/files#diff-a7b7ef841bb09679f68c5cab00300b0bL430) pass null as the second argument in case of a successful reading.